test/functional/legacy/094_visual_mode_operators_spec.lua

```diff
@@ -0,0 +1,153 @@
+      let [line2, col2] = searchpos('.\u', 'nW')
+      call setpos("'<", [0, line1, col1, 0])
+      call setpos("'>", [0, line2, col2, 0])
+      normal! gv
+    endfunction
+  ]])
+end
+
+describe('Visual mode and operator', function()
+  before_each(function()
+    clear()
+    source_user_functions()
+  end)
+
+  it('simple change in Visual mode', function()
+    insert([[
+      apple banana cherry
+      
+      line 1 line 1
+      line 2 line 2
+      line 3 line 3
+      line 4 line 4
+      line 5 line 5
+      line 6 line 6
+      
+      xxxxxxxxxxxxx
+      xxxxxxxxxxxxx
+      xxxxxxxxxxxxx
+      xxxxxxxxxxxxx]])
+
+    -- Exercise characterwise Visual mode plus operator, with count and repeat.
+    execute('/^apple')
+    feed('lvld.l3vd.')
+
+    -- Same in linewise Visual mode.
+    execute('/^line 1')
+    feed('Vcnewline<esc>j.j2Vd.')
+
+    -- Same in blockwise Visual mode.
+    execute('/^xxxx')
+    feed('<c-v>jlc  <esc>l.l2<c-v>c----<esc>l.')
+
+    -- Assert buffer contents.
+    expect([[
+      a y
+      
+      newline
+      newline
+      
+          --------x
+          --------x
+      xxxx--------x
+      xxxx--------x]])
+  end)
+
+  it('Visual mode mapping', function()
+    insert([[
+      KiwiRaspberryDateWatermelonPeach
+      JambuRambutanBananaTangerineMango]])
+
+    -- Set up Visual mode mappings.
+    execute('vnoremap W /\\u/s-1<CR>')
+    execute('vnoremap iW :<C-U>call SelectInCaps()<CR>')
+
+    -- Do a simple change using the simple vmap, also with count and repeat.
+    execute('/^Kiwi')
+    feed('vWcNo<esc>l.fD2vd.')
+
+    -- Same, using the vmap that maps to an Ex command.
+    execute('/^Jambu')
+    feed('llviWc-<esc>l.l2vdl.')
+
+    -- Assert buffer contents.
+    expect([[
+      NoNoberryach
+      --ago]])
+  end)
+
+  it('Operator-pending mode mapping', function()
+    insert([[
+      PineappleQuinceLoganberryOrangeGrapefruitKiwiZ
+      JuniperDurianZ
+      LemonNectarineZ]])
+
+    -- Set up Operator-pending mode mappings.
+    execute('onoremap W /\\u/<CR>')
+    execute('onoremap <Leader>W :<C-U>call MoveToCap()<CR>')
+    execute('onoremap iW :<C-U>call SelectInCaps()<CR>')
+
+    -- Do a simple change using the simple omap, also with count and repeat.
+    execute('/^Pineapple')
+    feed('cW-<esc>l.l2.l.')
+
+    -- Same, using the omap that maps to an Ex command to move the cursor.
+    execute('/^Juniper')
+    feed('g?\\WfD.')
+
+    -- Same, using the omap that uses Ex and Visual mode (custom text object).
+    execute('/^Lemon')
+    feed('yiWPlciWNew<esc>fr.')
+
+    -- Assert buffer contents.
+    expect([[
+      ----Z
+      WhavcreQhevnaZ
+      LemonNewNewZ]])
+  end)
+
+  -- Vim patch 7.3.879 addressed a bug where typing ":" (the start of an Ex
+  -- command) in Operator-pending mode couldn't be aborted with Escape, the
+  -- change operation implied by the operator was always executed.
+  it('patch 7.3.879', function()
```

Comment:
I reverted https://github.com/vim/vim/commit/88365df32d50048579d80ad86185d5f221484e51 locally, and this test failed :+1: